lec15-git-projects.Rmd

```diff
@@ -0,0 +1,415 @@
+which can be downloaded from the Mac App store. Be warned: Xcode is a very 
+large download (approximately 6 GB)! Install Xcode and Git should be
+ready for use.
+
+To install Git (and Bash) on **Windows**, download [Git for Windows](https://gitforwindows.org/) from its [website](https://github.com/git-for-windows/git/releases/tag/v2.19.1.windows.1). 
+At the time of writing, 2.19.1 is the most recent version. Download the `exe`
+file to get started. Git for Windows provides a program called Git Bash, which
+provides a bash interpreter that can also run Git commands. 
+
+To install Git for **Linux**, use the preferred package manager
+for your distribution and follow the instructions [listed here](https://git-scm.com/download/linux). 
+
+To test whether Git has successfully been installed, open a bash interpreter, type
+
+```
+git --version
+```
+
+and hit enter. If the interpreter returns a version number, Git has been
+installed.
+
+#### Bash Primer
+
+Before we can use Git in the bash interpreter, it's worth learning a
+few basic bash commands. All bash commands are run simply by typing them 
+into the interpreter and pressing enter. 
+
+```
+pwd
+```
+
+`pwd`, or 'print working directory', will return the folder bash
+is currently in. This is similar to the concept of the working directory in R
+that we discussed earlier in the course.
+
+```
+cd [folder]
+```
+
+`cd` changes the current working directory. This is useful in tandem with
+`~`, which is shorthand for 'home directory' in bash (usually 
+`/Users/[yourname]`). To navigate to your Desktop, the command would be 
+`cd ~/Desktop`. 
+
+`cd ..` will let you move back up one folder. If your current working
+directory is `/Users/[yourname]/Desktop`, for instance, `cd ..` will
+change it to `/Users/[yourname]`. 
+
+```
+ls
+```
+
+`ls` will return what files/folders are within the current 
+working directory. This is useful when you have just used `cd` to 
+change your current directory and need a quick reminder of what's
+in said directory. 
+
+```
+mkdir [foldername]
+```
+
+`mkdir`, or 'make directory', will create a few directory with
+the specified name. For instance, `mkdir thesis` will create a folder
+called `thesis`, which can then be entered with `cd thesis`.
+
+```
+nano [filename]
+```
+
+`nano` is a text editor, similar to Notepad on Windows, except that
+it opens right within the bash window. If the filename provided doesn't 
+refer to an existing file, `nano` will create a new file with that name. 
+Once you're done with `nano`, press Control + X to exit - `nano` will ask 
+whether you want to save your work, which can be answered by pressing either 
+`y` or `n`. Alternatively, Control + O followed by Enter will save your current 
+work, after which Control + X will exit without prompting.
+
+#### Getting started with Git
+
+First, we have to tell Git who we are. This is especially important
+when collaborating with Git and keeping track of who did what!
```

Comment:
Should mention that this only has to be done once and not every time you want to use git.